internal/neofs/neofs.go

```diff
@@ -277,6 +277,15 @@ func (x *NeoFS) CreateObject(ctx context.Context, prm layer.PrmObjectCreate) (oi
 		return oid.ID{}, fmt.Errorf("save object via connection pool: %w", err)
 	}
 
+	// Lock payload has been written to header directly, thus we shouldn't miss this payload too.
+	if data := obj.Payload(); len(data) > 0 {
```

Comment:
It's really unobvious this way.

Reply:
Removed all this construction and moved code upper to locks